optimizely/helpers/condition.py

```diff
@@ -153,6 +153,18 @@ def compare_user_version_with_target_version(self, target_version, user_version)
             - if the user version value format is not a valid semantic version.
         """
 
+        if not isinstance(target_version, string_types):
+            self.logger.warning(audience_logs.UNKNOWN_CONDITION_VALUE.format(self._get_condition_json(0), ))
+            return None
+
+        if not isinstance(user_version, string_types):
+            self.logger.warning(
+                audience_logs.UNEXPECTED_TYPE.format(
+                    self._get_condition_json(0), type(user_version), user_version
+                )
+            )
+            return None
```

Comment:
We don't need to put this here because
- We have already verified these in all evaluators that call this method. 
- This method doesn't have attribute name and index to log the error.